smartmeter/filesystem.cpp

```diff
@@ -0,0 +1,102 @@
+/**
+ * HaLiO software
+ * Copyright 2017 Patrick José Pereira <patrick@ibexcps.com>
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses/.
+ */
+
+#include <vector>
+
+#include <Arduino.h>
+
+#include "debug.h"
+#include "filesystem.h"
+
+fs::FS FileSystem::FS = SPIFFS;
+const char* FileSystem::_lockFileName = "/.fileSystem";
+
+FileSystem::FileSystem():
+    _fileSystemOk(false)
+{
+    FS.info(_fileSystemInfo);
+}
+
+void FileSystem::start()
+{
+    debug("Starting file system...");
+    if (!FS.begin()) {
+        _fileSystemOk = false;
+        error();
```

Comment:
wouldn't it be interesting if "error()" took an string argument, so we didn't need to to use call debug() before to explain what happened?